From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Wed, 15 Jun 2016 19:31:49 +0200
Subject: tools/cmake: fix parallel build with Make 4.2+

Avoid using the --parallel argument to the CMake bootstrap, as that doesn't
allow us to remove the -j argument. Instead, pass the HOST_JOBS arguments
in MAKEFLAGS.

Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>

Backport of LEDE 96db10752483cb07f822686482b44be6c21fbb5f

diff --git a/tools/cmake/Makefile b/tools/cmake/Makefile
index 08e2b028150dd4f496e3a8ebccf27bd49b6937c2..6c211def540a6a41012349d5979b919481e1bcec 100644
--- a/tools/cmake/Makefile
+++ b/tools/cmake/Makefile
@@ -18,10 +18,11 @@ HOST_CONFIGURE_PARALLEL:=1
 
 include $(INCLUDE_DIR)/host-build.mk
 
+HOST_CONFIGURE_CMD := MAKEFLAGS="$(HOST_JOBS)" $(BASH) ./configure
+
 HOST_CONFIGURE_VARS :=
 
 HOST_CONFIGURE_ARGS := \
-	--prefix=$(STAGING_DIR_HOST) \
-	$(if $(MAKE_JOBSERVER),--parallel="$(MAKE_JOBSERVER)")
+	--prefix=$(STAGING_DIR_HOST)
 
 $(eval $(call HostBuild))
